wlc/__init__.py

```diff
@@ -52,47 +51,51 @@ def __init__(self, key='', url=API_URL, config=None):
     @staticmethod
     def process_error(error):
         """Raise WeblateException for known HTTP errors."""
-        if hasattr(error, 'code'):
-            if error.code == 429:
+        if type(error) is HTTPError:
+            status_code = error.response.status_code
+
+            if status_code == 429:
                 raise WeblateException(
                     'Throttling on the server'
                 )
-            elif error.code == 404:
+            elif status_code == 404:
                 raise WeblateException(
                     'Object not found on the server '
                     '(maybe operation is not supported on the server)'
                 )
-            elif error.code == 403:
+            elif status_code == 403:
                 raise WeblateException(
                     'You don\'t have permission to access this object'
                 )
             raise WeblateException(
-                'HTTP error {0}: {1}'.format(error.code, error.reason)
+                'HTTP error {0}: {1}'.format(status_code, error.response.reason)
             )
 
-    def request(self, path, params=None, raw=False):
-        """Construct request object."""
+    def request(self, method, path, params=None, raw=False):
         if not path.startswith('http'):
             path = '{0}{1}'.format(self.url, path)
-        request = Request(path)
-        request.add_header('User-Agent', USER_AGENT)
-        request.add_header('Accept', 'application/json')
+
+        headers = {'user-agent': USER_AGENT, 'Accept': 'application/json'}
+
         if self.key:
-            request.add_header(
-                'Authorization',
-                'Token {}'.format(self.key)
-            )
+            headers['Authorization'] = 'Token {}'.format(self.key)
+
+        verify_ssl = not path.startswith('https://127.0.0.1')
```

Comment:
I'd either completely remove this or make this configurable. Hardcoding one specific URL here seems wrong to me...

Reply:
Me too. Any suggestions?

Reply:
This is only required while testing. That being said, some may want to use self-signed certificates.